port/unix/omrsysinfo.c

```diff
@@ -128,7 +133,15 @@
 #endif
 
 #if defined(LINUX) && !defined(OMRZTPF)
+#if defined(__GLIBC__)
 #include <linux/magic.h>
+#else
+/* Linux distros which doesn't use glibc toolchain like
+ * Alpine Linux the magic.h file is not available
+ * Random number for TMPFS_MAGIC variable is taken from other distros
+ */
+#define TMPFS_MAGIC 0x01021994
```

Comment:
What breaks if this magic number isn't here?

I'm confused that musl doesn't define it and yet somehow us defining it is the right solution to the problem caused by TPFS_MAGIC not being there...

As it stands, I don't think I want code "taken from other distros" as the comment states.

Reply:
These sorts of magic numbers are fine; they're returned by the `statfs()` call and are well documented, e.g. in `man statfs`. The header in question is not part of any C library, it's part of the Linux kernel so I'm not sure why it wouldn't exist on a musl system. Is there some sort of headers package on musl that you can/should install? If so, it might be better to rewrite this as `#ifndef TMPFS_MAGIC ...` rather than looking for `__GLIBC__` but even guarding the include with `__GLIBC__` might be problematic because it doesn't guarantee its presence.

The code in question seems to call `statfs()` on the typical cgroups root mount point and I guess wants to confirm that it is a tmpfs based file system because that's what it usually is, but I don't know if that's really a requirement. I don't see why it couldn't be any other kind of FS, in which case this could would break.

Reply:
TMPFS_MAGIC is one of the file system type. It is clearly documented in man page of statfs. http://man7.org/linux/man-pages/man2/statfs.2.html.  I will correct the comment by clearly stating that "It is one of the file system type and it is documented in statfs system call man page"